### DISCUSSION TITLE
Update failing because of RLS for SELECT

### DISCUSSION THREAD

**Discussion by iddogino** [supabase/supabase#18684] [Questions]
I have a table `Account(id, status, owner)` with the following RLS policies:

**For SELECT:**
```SQL
CREATE POLICY "Allow users to select active accounts" ON "public"."Account"
AS PERMISSIVE FOR SELECT
TO public
USING ((owner = auth.uid()) AND (status = 'active'))
```

**For UPDATE:**
```SQL
CREATE POLICY "Allow users to update their own accounts" ON "public"."Account"
AS PERMISSIVE FOR UPDATE
TO public
USING ((owner = auth.uid()))
WITH CHECK ((owner = auth.uid()))
```

Selecting works fine, but when I try updating a record to `status='deleted'` I get an RLS violation error. I am using the JS SDK (latest version) for this update with this code:
```javascript
await supabase.from("Account").update({"status":"deleted"}).eq('id', id)
```

If I change the SELECT policy and remove the condition `AND (status = 'active')` -- then the update works. So it seems that this is what's triggering the issue. However, as I am not selecting with the update (and it should return minimal by default), I don't understand why it's causing this issue. What am I missing here?

**Comment by GaryAustin1** [comment]
Postgres will not allow this.  You have to meet select/using policies both before and after.
You will need to use a security definer rpc function to do the update of the status.

**Reply by GaryAustin1** [reply]
Just dropping this here.  Not the best example of the issue I've encountered here but very detailed. 
https://github.com/orgs/supabase/discussions/9536
